### PR TITLE
add bst test

### LINKED ISSUE #40: Need to write tests for bst.c
Need to write tests for the binary search tree implementation in bst.c and bst.h

### PR DESCRIPTION
**Pull request by wanghenshui**
fix https://github.com/f-prime/fist/issues/40

### CODE REVIEW

**Comment by 00-matt** [comment]
We could refactor bst to store anything (`void *`?) instead of just `command_handler_t`s to try and clean up this code

**Comment by wanghenshui** [comment]
> We could refactor bst to store anything (`void *`?) instead of just `command_handler_t`s to try and clean up this code


wait for it

File: fist/tests.c [hunk 1]
```diff
@@ -389,6 +390,44 @@ static char *test_serialize_hmap() {
...
+}
+void test_helper_handler3() {
+
+}
+int* test_helper_handler4() {
+    return NULL;
+}
+static char *test_insert_and_search_bst(){
+    struct bst_node *root = bst_create("Hello World", NULL);
+    bst_insert(&root, "BSTn",test_helper_handler1);
```
Comment (00-matt) [inline fist/tests.c]: Now that bst stores `void *`, you could just pass in an int or string or something instead of making `test_helper_handler{1..4}`

Comment (wanghenshui) [reply fist/tests.c]: Clean up

**Review by 00-matt — APPROVED** [review]
Looks good, everything seems to work and the tests pass :tada:

**Comment by f-prime** [comment]
Looks good! Thanks for reviewing Matt. Merging.

**Pull request merged**